kpops/component_handlers/kafka_connect/kafka_connect_handler.py

```diff
@@ -44,57 +44,77 @@ def create_connector(
         :param dry_run: If the connector creation should be run in dry run mode.
         """
         if dry_run:
-            self.__dry_run_connector_creation(connector_config)
+            await self.__dry_run_connector_creation(connector_config)
         else:
             try:
-                timeout(
-                    lambda: self._connect_wrapper.get_connector(connector_config.name),
+
+                async def get_connector_locally():
+                    return await self._connect_wrapper.get_connector(
+                        connector_config.name
+                    )
+
+                await timeout(
+                    get_connector_locally(),
                     secs=self._timeout,
                 )
 
-                timeout(
-                    lambda: self._connect_wrapper.update_connector_config(
+                async def update_connector_locally():
+                    await self._connect_wrapper.update_connector_config(
                         connector_config
-                    ),
+                    )
+
+                await timeout(
+                    update_connector_locally(),
                     secs=self._timeout,
                 )
 
             except ConnectorNotFoundException:
-                timeout(
-                    lambda: self._connect_wrapper.create_connector(connector_config),
+
+                async def create_connector_locally():
+                    await self._connect_wrapper.create_connector(connector_config)
+
+                await timeout(
+                    create_connector_locally(),
                     secs=self._timeout,
                 )
 
-    def destroy_connector(self, connector_name: str, *, dry_run: bool) -> None:
+    async def destroy_connector(self, connector_name: str, *, dry_run: bool) -> None:
         """Delete a connector resource from the cluster.
 
         :param connector_name: The connector name.
         :param dry_run: If the connector deletion should be run in dry run mode.
         """
         if dry_run:
-            self.__dry_run_connector_deletion(connector_name)
+            await self.__dry_run_connector_deletion(connector_name)
         else:
             try:
-                timeout(
-                    lambda: self._connect_wrapper.get_connector(connector_name),
+
+                async def get_connector_locally():
+                    return await self._connect_wrapper.get_connector(connector_name)
+
+                await timeout(
+                    get_connector_locally(),
                     secs=self._timeout,
                 )
 
-                timeout(
-                    lambda: self._connect_wrapper.delete_connector(connector_name),
+                async def delete_connector_locally():
+                    await self._connect_wrapper.delete_connector(connector_name)
+
+                await timeout(
+                    delete_connector_locally(),
```

Comment:
why does it need to be wrapped in another async function, can't you pass `self._connect_wrapper.delete_connector(connector_name)`?